.github/workflows/scanner-versioned-definitions-update.yaml

```diff
@@ -248,20 +248,20 @@ jobs:
   upload-definitions:
     needs:
     - build-and-run
-    if: github.event_name == 'schedule' || github.event_name == 'workflow_dispatch'
+    if: ${{ (failure() || success()) && (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') }}
     runs-on: ubuntu-latest
     steps:
     # Checkout to run ./.github/actions/download-artifact-with-retry
     - uses: actions/checkout@v4
       with:
         fetch-depth: 0
 
-    - uses: ./.github/actions/download-artifact-with-retry
+    - uses: actions/download-artifact@v4
       with:
-        # Only download vulnerability bundles.
-        # This avoids accidentally downloading nvd.zip.
+        # Vulnerability bundles are named `artifact_${{ env.SCANNER_BUNDLE_VERSION }}`.
         pattern: "artifact_*"
         path: downloaded_artifacts
+        if-no-files-found: error
```

Comment:
Should we adjust https://github.com/stackrox/stackrox/blob/master/.github/actions/download-artifact-with-retry/action.yaml instead? Maybe add `if-no-files-found: error` to the last line of the shared action

Reply:
Turns out `if-no-files-found` is part of https://github.com/actions/upload-artifact not https://github.com/actions/download-artifact

I will do a manual check.

Reply:
Do you have an example of this passing even though no files were downloaded?